src/inmanta/plugins.py

```diff
@@ -266,18 +266,35 @@ def get_location(self) -> Optional[Location]:
 
 @dataclass(frozen=True)
 class ModelType:
+    """
+    Dataclass used with typing.Annotated to represent Inmanta model types in Python code.
+
+    If we want to represent "std::Entity" as "typing.Any" in our plugins we could define the following type:
+
+    Entity: typing.TypeAlias = typing.Annotated[typing.Any, ModelType["std::Entity"]]
+
+    or even
```

Comment:
I see no reason to show the two syntaxes here, I would simply use the new one (`type` keyword).